cfme/services/myservice.py

```diff
@@ -132,31 +211,33 @@ def retire(self):
             assert vm_state in ['off', 'unknown']
 
     def retire_on_date(self, retirement_date):
-        navigate_to(self, 'SetRetirement')
-        fill(retirement_form, {'retirement_date': retirement_date},
-             action=form_buttons.save)
-        navigate_to(self, 'Details')
+        view = navigate_to(self, 'SetRetirement')
+        view.retirement_date.fill(retirement_date)
+        view.save_button.click()
+        view = navigate_to(self, 'Details')
         wait_for(
-            lambda: InfoBlock.text('Lifecycle', 'Retirement State') == 'Retiring',
+            lambda: view.lifecycle_detail.get_text_of("Retirement State") == 'Retiring',
             fail_func=tb.refresh,
             num_sec=5 * 60,
             delay=5,
             message='Service Retirement'
         )
         # wait for any vms to shutdown, this method leaves the page on the VM details
-        vm_state = self.wait_for_vm_retire()
-        assert vm_state in ['off', 'unknown']
+        if self.vm_name is not None:
+            vm_state = self.wait_for_vm_retire()
+            assert vm_state in ['off', 'unknown']
```

Comment:
Later I would like to see something like this, but that will require a change in the model and is not subject of this pr:
```python
for vm in self.vms:
    vm.some_method_that_waits_for_power_state(vm.RETIRED)
```